fpdf/fpdf.py

```diff
@@ -2506,8 +2489,17 @@ def _render_styled_cell_text(
                     self.font_size,
                     link,
                 )
-        if s:
-            self._out(s)
+        if sl:
+            # If any PDF settings have been left modified, wrap the line in a local context.
+            if (
+                current_font_style != self.font_style
+                or self.fill_color != self.text_color
+            ):
+                s = f"q {' '.join(sl)} Q"
+                self._out(s)
+            else:
+                s = " ".join(sl)
+                self._out(s)
```

Comment:
This line is shared between the 2 parts of the `if` and could be de-indented

Reply:
yup.